pkg/factory/factory.go

```diff
@@ -114,6 +115,13 @@ func (rf *resourceFactory) GetResourcePool(rc *types.ResourceConfig, filteredDev
 		if len(filteredDevice) > 0 {
 			if _, ok := filteredDevice[0].(types.PciNetDevice); ok {
 				rPool = netdevice.NewNetResourcePool(rc, apiDevices, devicePool)
+				// check /dev/infiniband exist if rdma is required
+				selector, _ := rc.SelectorObj.(*types.NetDeviceSelectors)
+				if selector.IsRdma {
+					if _, err := os.Stat("/dev/infiniband"); err != nil {
```

Comment:
If `/dev/infiniband` doesn't exist and `isRdma` is true, the `filteredDevice` would be zero.
I think this code line won't be reached.

Reply:
Maybe add the rdma check in `GetFilteredDevices` when filtering devices or simply in `validConfigs`.

Reply:
moved to `GetFilteredDevices`